File: java/src/main/java/com/test/java/animals/CatIF.java
package com.test.java.animals;

import com.hubspot.immutables.style.HubSpotStyle;
import org.immutables.value.Value;

@Value.Immutable
@HubSpotStyle
public interface CatIF extends Animal {
    @Value.Default
    default String getNoise() {
        return "Meow";
    }

    boolean getLikesMilk();
}

File: kotlin/src/main/kotlin/com/test/kotlin/animals/CatIF.kt
package com.test.kotlin.animals

import com.hubspot.immutables.style.HubSpotStyle
import org.immutables.value.Value

@Value.Immutable
@HubSpotStyle
interface CatIF : Animal {
    override val noise: String
        @Value.Default
        get() = "Meow"

    val likesMilk: Boolean
}
